### PR TITLE
Minor cleanups to Fido

### PR DESCRIPTION
**Pull request by Cadair**
I found these while working on the DKIST client.

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

File: sunpy/net/base_client.py [hunk 1]
```diff
@@ -207,6 +207,36 @@ def unhide_columns(self):
...
+        -------
+        new_table : QueryResponseTable
+            A sliced version of this table instance so that the columns are
+            reordered.
+        """
+        all_cols = list(self.colnames)
+        first_names = [n for n in first_columns if n in all_cols]
+        extra_cols = [col for col in all_cols if col not in first_names]
+        all_cols = first_names + extra_cols
+        new_table = self[[col for col in all_cols if data[col] is not None]]
```
Comment (nabobalis) [inline sunpy/net/base_client.py:232]: Data is not defined.

Comment (Cadair) [reply sunpy/net/base_client.py:232]:
```suggestion
        new_table = self[[col for col in all_cols if self[col] is not None]]
```

Comment (Cadair) [inline sunpy/net/base_client.py:232]:
```suggestion
        new_table = self[[col for col in all_cols if self[col] is not None]]
```

File: sunpy/net/base_client.py [hunk 2]
```diff
@@ -207,6 +207,36 @@ def unhide_columns(self):
...
 
+    def _reorder_columns(self, first_columns, remove_empty=True):
+        """
+        Generate a new version of this table with ``first_columns`` at the start.
+
+        Parameters
+        ----------
+        first_columns : list
+           The column names to put at the start of the table.
+
```
Comment (nabobalis) [inline sunpy/net/base_client.py:218]:
```suggestion
```

File: sunpy/net/base_client.py [hunk 3]
```diff
@@ -207,6 +207,36 @@ def unhide_columns(self):
...
+        remove_empty : bool, optional
+           Remove columns where all values are `None`.
+
```
Comment (nabobalis) [inline sunpy/net/base_client.py:221]:
```suggestion
           Defaults to ``True``.
```

**Review by dstansby — APPROVED** [review]

**Comment by dstansby** [comment]
Online failure is unrelated (but weird, not sure where it's come from...)

**Pull request merged**